docs/src/cirrus/70_statedb.rst

```diff
@@ -1,14 +1,100 @@
+
+Additionally as a NoSQL database, Dynamo DB does not require a predefined
+schema and permits diferent items to have different attributes while the rigid
+schema of relational databases means there can be no variaton of the data
+stored in a given table.  In fact each entry and its attributes in the cirrus
+state DB is completely independant of other items in the state DB.
+
+Managed Serverless Service
+--------------------------
+
+As a managed AWS service Dynamo DB handles provisoning and maintaining the
+underlying storage and scaling infrastructure for your tables as your data
+scales up or down.  This allows cirrus to focus on the business logic of state
+management.  Additonally, Dynamo DB is optimized by AWS for rapid read/writes
+at any scale.
+
 Schema
 ------
+While Dynamo DB does not necessitate a predefined schema like a relational
+databse, there are attributes that are required for core cirrus functionality.
+Users may add additional fields if necessary.  Because Dynamo DB does not
+require a predefined schema users may add additonal attributes as needed.
+
+Required Fields:
+These fields are required for out of the box functionality of cirrus
+
+* ``collections_workflow`` (*string*):  a unique "partition key" constructed from a cirrus payload's ``payload_id``
+* ``itemIDs`` (*string*): a unique ID field extrated from a payload ID
+* ``created`` (*string*): UTC time when record was created
+* ``executions`` (*list[string]*): ARNs of state machine executions.  May have multiple records in this field if a payload is submitted multiple time, or part of chained workflows
+* ``state_updated`` (*string*): Concatenated string of state + UTC of last updated
+* ``updated`` (*string*): UTC time when the record was most recently updated
+
+State DB and Cirrus CLI
+-----------------------
+
+Selected Cirrus CLI commands interact with the state DB.
+
+The ``get-payloads`` command takes in query parameters to retrieve input payloads in bulk, returned as new line delimited JSON.
+
+Input query parameters are used as filters against the state DB to retrieve
+records matching certain criteria, like entires with a ``FAILED`` workflow
+status that occured in the past week.  These returned state DB records are used
+to retrieve input payloads from the S3 payload bucket which are returned to the
+user.
+
+One use for these returned bulk payloads is to pipe them into another
+cirrus CLI command to rerun failed payloads, perhaps in the event of a third
+party service failure that resulted in failed workflow executons.
+
+More information about the ``get-payloads`` command can be found in the CLI
+documentation
 
-Using cirrus-lib to interact with state database
-------------------------------------------------
+Deleting state DB items
+-----------------------
 
-Using the Cirrus API
---------------------
+There is a ``delete_item`` method in the "StateDB" class that can be used to
+delete a given item based on using ``payload_id``.
```

Comment:
Really only wanted to move the discouragement and warning to the beginning of this section.  But, I think `StateDB` gets double-back-quotes, right?   And I tried fixing up the tail end of the sentence, which gets confusing:

```suggestion
The deletion of records from the StateDB is strongly discouraged, and almost
certainly unnecessary.  For the unlikely case it is required, there is a 
``delete_item`` method of the ``StateDB`` class.  That can be used to delete 
a record, based on its ``payload_id``.
```